### DISCUSSION TITLE
Calculated field from another field

### DISCUSSION THREAD

**Discussion by yangetze** [Megabit/Blazorise#2833] [Q&A] [answered]
Hi community, I have a question. 

I have two numeric fields (field1, field2). 
field2 is calculated as from field1. I need to show that result in the input everytime there's a change in field1.
```
<NumericEdit TValue="decimal" Value="@NewInput.Field1" ValueChanged="@OnField1Changed"></NumericEdit>

<NumericEdit TValue="decimal" @bind-Value="@NewInput.Field2" ReadOnly="true"</NumericEdit>

```

In the class I have this
```
 public void OnField1Changed(decimal valField1) {
            NewInput.Field2 =
                SomeAppService.Caculate(valField1, 100M)
                .Result;
        }
```

But I did't see any result. 
I have something wrong? I test the method OnField1Changed on my backeand and it works. 
Is there another way to solve this?

Blazorise version: 0.9.3-preview6

Thx in Advanced!

**Comment by stsrki** [comment, marked as answer]
**Try this**:

```cshtml
<NumericEdit TValue="decimal" Value="@Field1" ValueChanged="@OnField1Changed" />

<NumericEdit TValue="decimal" Value="@Field2" ReadOnly="true" />

@code{
    decimal Field1;

    decimal Field2;

    Task OnField1Changed( decimal newValue )
    {
        Field1 = newValue;

        Field2 = Field1 * 0.25m;

        return Task.CompletedTask;
    }
}
```

I think you only missed `Task` on the method `OnField1Changed`. It must be available so that Blazor can update the tree.

Also, you don't need `@bind-` for the readonly field.

**Reply by yangetze** [reply]
It works! 
Thankkks!

**Reply by stsrki** [reply]
Cool :) 

Can you mark the answer? thanks!

**Reply by yangetze** [reply]
I'm trying to do that since yesterday but I didn't work 
[Check](https://github.com/github/feedback/discussions/2903)
But i would add this as a task to mark later ;)
